File: ClientApp/src/app/features/reservations/classes/resources/form/passenger-read-resource.ts
import { Guid } from "guid-typescript"

export class PassengerReadResource {

    id: number
    
    reservationId: Guid

    occupant: { id: number, description: string }
    nationality: { id: number, description: string }
    gender: { id: number, description: string }

    lastname: string
    firstname: string
    birthdate: string
    specialCare: string
    remarks: string
    isCheckedIn: boolean

}

File: API/Features/Reservations/Resources/Passengers/PassengerReadResource.cs
using System;
using API.Infrastructure.Classes;

namespace API.Features.Reservations {

    public class PassengerReadResource {

        public int Id { get; set; }

        public Guid ReservationId { get; set; }

        public string Lastname { get; set; }
        public string Firstname { get; set; }
        public string Birthdate { get; set; }
        public string Remarks { get; set; }
        public string SpecialCare { get; set; }
        public bool IsCheckedIn { get; set; }

        public SimpleResource Nationality { get; set; }
        public SimpleResource Gender { get; set; }

    }

}